### PR TITLE
XW-514 Convert utils to ES6

### PR DESCRIPTION
**Pull request by rogatty**
https://wikia-inc.atlassian.net/browse/XW-514

### CODE REVIEW

File: front/scripts/baseline/mercury/utils/buildUrl.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+ * @param {string} wiki - The new wiki, which may contain a language prefix; for example, "glee" or "es.walkingdead"
+ * @returns {string} New host
+ */
+export function replaceWikiInHost(host, wiki) {
+	let match;
+
+	// (1) Sandbox, preview, or verify hosts on wikia.com
+	if ((match = host.match(/^(sandbox-.+?|preview|verify)\.(.+?)\.wikia\.com($|\/|:)/)) !== null) {
+		host = host.replace(`${match[1]}.${match[2]}`, `${match[1]}.${wiki}`);
+		// (2) Production wikia.com
```
Comment (Warkot) [inline front/scripts/baseline/mercury/utils/buildUrl.js]: I know it's probably due to auto formatting – comment is still inside the curly braces, but logically it concerns `else if` in the next line, so I would keep the old indentation (just one tab like on line 104).

File: front/scripts/baseline/mercury/utils/loadScript.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/**
+ * @throws URIError exception
+ * @param {*} error
+ * @returns {void}
+ */
+function loadError(error) {
+	const scriptTag = error.target;
+
+	throw new URIError(`The script ${scriptTag.src} is not accessible.`);
```
Comment (Warkot) [inline front/scripts/baseline/mercury/utils/loadScript.js]: Why not just `error.target.src`?

File: front/scripts/baseline/mercury/utils/buildUrl.js (hunk 1 above)
Comment (rogatty) [inline front/scripts/baseline/mercury/utils/buildUrl.js]: ESLint won't like it and me neither. I think we should move all comments one level below, in other words put them inside correlated conditions.

File: front/scripts/baseline/mercury/utils/buildUrl.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	const mediawikiDomain = prop('mediawikiDomain'),
+		host = context.location.host;
+
+	let url;
+
+	if (!urlParams.protocol) {
+		urlParams.protocol = 'http';
+	}
+
+	url = `${urlParams.protocol}://`;
```
Comment (tomasznapieralski) [inline front/scripts/baseline/mercury/utils/buildUrl.js]: combine with `let url` above

File: front/scripts/baseline/mercury/utils/state.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+		value: !mutable && !isPrimitive(val) ? Object.freeze(val) : val
+	};
+
+	if (mutable) {
+		properties.configurable = true;
+		properties.enumerable = true;
+		properties.writable = true;
+	}
+
+	for (i = 0; i < parts.length; i++) {
```
Comment (tomasznapieralski) [inline front/scripts/baseline/mercury/utils/state.js]: `let i`

File: front/scripts/baseline/mercury/utils/state.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+/**
+ * @param {string} str
+ * @param {string|object} ns
+ * @param {*} val
+ * @param {boolean} [mutable]
+ * @returns {*}
+ */
+function namespacer(str, ns, val, mutable) {
+	let parts,
+		i,
```
Comment (tomasznapieralski) [inline front/scripts/baseline/mercury/utils/state.js]: remove `i`

File: front/scripts/baseline/mercury/utils/state.js [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+
+	if (!ns) {
+		ns = window;
+	}
+
+	if (typeof ns === 'string') {
+		ns = window[ns] = window[ns] || {};
+	}
+
+	properties = {
```
Comment (tomasznapieralski) [inline front/scripts/baseline/mercury/utils/state.js]: `const properties`

File: front/scripts/baseline/mercury/utils/state.js [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+ * @param {string} str
+ * @param {string|object} ns
+ * @param {*} val
+ * @param {boolean} [mutable]
+ * @returns {*}
+ */
+function namespacer(str, ns, val, mutable) {
+	let parts,
+		i,
+		properties;
```
Comment (tomasznapieralski) [inline front/scripts/baseline/mercury/utils/state.js]: remove `properties`

File: front/scripts/baseline/mercury/utils/state.js [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+function _getProp(key) {
+	const parts = key.split('.');
+
+	let value,
+		i;
+
+	if (parts.length > 1) {
+		i = 0;
+		value = __props__;
+		while (i < parts.length) {
```
Comment (tomasznapieralski) [inline front/scripts/baseline/mercury/utils/state.js]: convert to `for`

File: front/scripts/baseline/mercury/utils/state.js [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+
+	if (typeof mutable !== 'boolean') {
+		throw new Error('Argument 2, mutable, must be a boolean value');
+	}
+
+	if (typeof value === 'string' || !keys.length) {
+		throw new Error(`Unable to set properties with the supplied value: ${value} (of type ${typeof value})`);
+	}
+
+	while (l > -1) {
```
Comment (tomasznapieralski) [inline front/scripts/baseline/mercury/utils/state.js]: `for`

File: front/scripts/baseline/mercury/utils/state.js [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+		curVal = value[keys[l]];
```
Comment (tomasznapieralski) [inline front/scripts/baseline/mercury/utils/state.js]: const

File: front/scripts/baseline/mercury/utils/state.js [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+/**
+ * @param {string} str
+ * @param {*} obj
+ * @returns {*}
+ */
+export function provide(str, obj) {
+	if (typeof str !== 'string') {
+		throw Error('Invalid string supplied to namespacer');
+	}
+	return namespacer(str, 'Mercury', obj, true);
```
Comment (tomasznapieralski) [inline front/scripts/baseline/mercury/utils/state.js]: space above

File: front/scripts/mercury/utils/browser.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+	const ua = window.navigator.userAgent;
+
+	let system;
+
+	if (ua.match(/iPad|iPhone|iPod/i) !== null) {
+		system = 'ios';
+	} else if (ua.match(/Android/i) !== null) {
+		system = 'android';
+	}
+	return system;
```
Comment (tomasznapieralski) [inline front/scripts/mercury/utils/browser.js]: extra line above

File: front/scripts/baseline/mercury/utils/loadScript.js [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+
+	script.type = 'text\/javascript';
+	script.onerror = loadError;
+
+	if (fOnload) {
+		script.onload = fOnload;
+	}
+
+	document.body.appendChild(script);
+	script.src = src;
```
Comment (Warkot) [inline front/scripts/baseline/mercury/utils/loadScript.js]: I know we shouldn't work on changing logic, but I believe that one operation on live DOM could be avoided by moving this line above the preceding one.

File: front/scripts/mercury/utils/dateTime.js [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+	timeDiff = Math.floor(timeDiff / 30);
+	if (timeDiff < 12) {
+		return {
+			type: interval.month,
+			value: timeDiff
+		};
+	}
+	// years
+	timeDiff = Math.floor(timeDiff / 12);
+	return {
```
Comment (tomasznapieralski) [inline front/scripts/mercury/utils/dateTime.js]: extra line

File: front/scripts/mercury/utils/variantTesting.js [hunk 15]
```diff
@@ -0,0 +1,152 @@
...
+ * Integrates Optimizely with Universal Analytics
+ *
+ * @param {Array} dimensions
+ * @returns {Array}
+ */
+export function integrateOptimizelyWithUA(dimensions) {
+	const optimizely = window.optimizely,
+		activeExperiments = this.getActiveExperimentsList();
+
+	let dimension,
```
Comment (tomasznapieralski) [inline front/scripts/mercury/utils/variantTesting.js]: declare inside `if` block

**Pull request merged**